_posts/2020-12-06-converting-your-laptop-to-a-janus-modem-using-unetaudio.md

```diff
@@ -0,0 +1,227 @@
+
+```
+> phy[1]
+« PHY »
+
+[org.arl.unet.phy.PhysicalChannelParam]
+  dataRate ⤇ 70.588234
+  errorDetection ⤇ true
+  fec = 1
+  fecList ⤇ [ICONV2]
+  frameDuration ⤇ 2.04
+  frameLength = 18
+  janus = false
+
+[org.arl.yoda.FhbfskParam]
+  chiplen = 1
+  fmin = 9520.0
+  fstep = 160.0
+  hops = 13
+  scrambler = 0
+  sync = true
+  tukey = true
+```
+
+Similarly, typing `phy[3]` will list the parameters of JANUS frame.
+
+```
+> phy[3]
+« PHY »
+
+[org.arl.unet.phy.PhysicalChannelParam]
+  dataRate ⤇ 58.181816
+  errorDetection ⤇ true
+  fec = 1
+  fecList ⤇ [ICONV2]
+  frameDuration ⤇ 1.1
+  frameLength = 8
+  janus = true
+
+[org.arl.yoda.FhbfskParam]
+  chiplen = 1
+  fmin = 9520.0
+  fstep = 160.0
+  hops = 13
+  scrambler = 0
+  sync = true
+  tukey = true
+```
+
+The first difference you will notice between the three is the `phy.janus` parameter which is set to `false` for the CONTROL channel and `true` for JANUS channel. The settings `phy.janus` &  `modulation = fhbfsk` sets up the modulation for JANUS.
+
+The other key parameters to take note of:
+- `fmin = 9520.0`, `fstep = 160.0` and `hops = 13` are the modulation parameters to setup `fhbfsk` as required by JANUS.
+- `fec = 7` chooses `ICONV2` from the `fecList`, as required by JANUS
+- `threshold = 0.0` indicates that reception of JANUS frames is disabled
+
+Notice how they are different between the CONTROL channel and the JANUS channel. As a result, the data rate provided by the CONTROL channel is different from that of the JANUS channel.
+
+To enable JANUS reception, you need to set:
+```
+> phy[3].threshold = 0.3
+```
+
+> NOTE: To avoid detection for other type of frames, you can disable the other two channels by setting the `threshold` to zero.
+
+Now comes the most exciting part – communicating between two computers using JANUS with Unet audio. First you need to setup you receiving computer to generate notification after a reception by typing:
+
+```
+subscribe phy
+```
+This will show notifications from `phy` on the shell.
+
+Next, you can make a JANUS transmission by typing:
+
+```
+> phy << new TxJanusFrameReq()
+AGREE
+phy >> TxFrameNtf:INFORM[type:#3 txTime:127730682]
+```
+
+This will transmit a broadcast JANUS frame.
+
+On the receiving computer, you should see:
+
+```
+phy >> RxFrameStartNtf:INFORM[type:#3 rxTime:127730375 rxDuration:1100000 detector:0.96]
+phy >> RxJanusFrameNtf:INFORM[type:#3 classUserID:0 appType:0 appData:0 mobility:false canForward:true txRxFlag:true rxTime:127730374 rssi:-44.2 cfo:0.0]
+```
+
+> NOTE: The JANUS frame format doesn’t contain user data by default, and so the MTU is 0 (`frameLength` of 8 as defined in JANUS specifications). JANUS allows vendors to provide their own “cargo”, and so UnetStack lets you increase the `frameLength` to make space for user data (`MTU`).
+
+
+If you would like to add a payload, you can increase the `phy.frameLength` parameter (default is 8). Let us transmit 1 byte of data.
```

Comment:
Can we use 10 bytes as an example? That way we can use `phy << new TxJanusFrameReq(data:[1,2])`. That makes it easier to understand that `[1,2]` is an array. Otherwise, it might be a little confusing.

Reply:
Agree!!! Video may need to fix this too?

Reply:
Added as an additional line to the blog instead of replacing it so that I don't have to change the video.